localstack/services/dynamodb/provider.py

```diff
@@ -491,8 +501,18 @@ def update_table(
             is_no_update_error = (
                 e.code == "ValidationException" and "Nothing to update" in e.message
             )
-            if is_no_update_error and update_table_input.get("ReplicaUpdates"):
-                table_name = update_table_input.get("TableName")
+            if is_no_update_error and not list(
+                {"TableClass", "ReplicaUpdates"} & set(update_table_input.keys())
+            ):
+                raise
+
+            table_name = update_table_input.get("TableName")
+
+            if update_table_input.get("TableClass"):
+                table_definitions = DynamoDBRegion.get().table_definitions.get(table_name)
```

Comment:
I think we should hedge against None values here.
Either:
```
table_definitions = DynamoDBRegion.get().table_definitions.setdefault(table_name, {})
```

... or we could define `table_definitions` in `DynamoDBRegion` as a `defaultdict`:
```
self.table_definitions = defaultdict(dict)
```